### PR TITLE
Generate nightlies from main branches

### PR DESCRIPTION
**Pull request by chapulina**
Solves #319 

Unblocks PRs tied to #331 which require nightlies for GitHub actions CI.

### CODE REVIEW

**Comment by chapulina** [comment]
Generated new nightly scheduler jobs using this branch:

https://build.osrfoundation.org/job/_dsl_ignition_collection/281/

Note how the nightly scheduler has all libraries populated now:

https://build.osrfoundation.org/job/ignition-edifice-nightly-scheduler/build?delay=0sec

Ran nightly scheduler using this branch and a few releases failed, looking into it:

https://build.osrfoundation.org/job/ignition-edifice-nightly-scheduler/37/console

**Comment by chapulina** [comment]
I think the failures were because the debbuild jobs didn't exist. 5c67c3f  should address that.

Generated debbuild jobs using this branch:

https://build.osrfoundation.org/job/_dsl_ignition/1059/

Retriggered nightly scheduler:

https://build.osrfoundation.org/job/ignition-edifice-nightly-scheduler/38/console

It's not there yet. Looking into it.

**Comment by chapulina** [comment]
Some libraries were missing `-release` repos. I created those, except for `ign-cmake3` and `ign-tools2` because  `new_ignition_release_repos.bash` doesn't work well for them: https://github.com/ignition-tooling/release-tools/pull/310#issuecomment-718925618.

Some release repos has a `master` branch instead of `main`, and the nightly release didn't like that. See #335 for future release repos. I manually fixed the ones that had already been created a while back.

Finally, the last issue is related to how `ign-transport10` has 2 digits in the version number. I haven't looked into what needs updating for that.

Here's the latest run:

https://build.osrfoundation.org/job/ignition-edifice-nightly-scheduler/40/console

File: jenkins-scripts/dsl/sdformat.dsl [hunk 1]
```diff
@@ -4,7 +4,7 @@ import javaposse.jobdsl.dsl.Job
 def sdformat_supported_branches = [ 'sdformat4', 'sdformat6', 'sdformat8' , 'sdformat9', 'sdformat10' ]
 def sdformat_gz11_branches = [ 'sdformat8', 'sdformat9', 'sdformat10', 'master' ]
 // nightly and prereleases
-def extra_sdformat_debbuilder = [  ]
+def extra_sdformat_debbuilder = [ 'main' ]
```
Comment (j-rivero) [inline jenkins-scripts/dsl/sdformat.dsl:7]: uh, this is going to create `main-debbuilder` job which is something we probably don't want. Which was your intention here?

**Review by j-rivero — CHANGES_REQUESTED** [review]

File: jenkins-scripts/dsl/sdformat.dsl (hunk 1 above)
Comment (chapulina) [reply jenkins-scripts/dsl/sdformat.dsl:7]: Oh! I was trying to create an `sdformat11-debbuilder`, from the `main` branch :upside_down_face:

Comment (chapulina) [reply jenkins-scripts/dsl/sdformat.dsl]: Reverted in f720f98

**Review by j-rivero — APPROVED** [review]

**Comment by scpeters** [comment]
> Finally, the last issue is related to how `ign-transport10` has 2 digits in the version number. I haven't looked into what needs updating for that.

I think I know what's causing this

**Comment by scpeters** [comment]
> > Finally, the last issue is related to how `ign-transport10` has 2 digits in the version number. I haven't looked into what needs updating for that.
> 
> I think I know what's causing this

I think it's this logic here:

https://github.com/ignition-tooling/release-tools/blob/f720f98cc467132b47207e22ba1bd2fc313bdecb/jenkins-scripts/dsl/ignition_collection.dsl#L593-L597

**Comment by j-rivero** [comment]
> I think it's this logic here:

right. I've patched it in 5ff00ab. Testing run here: https://build.osrfoundation.org/job/ignition-edifice-nightly-scheduler/51/console . Fails for transport10-release (no main branch) and sdformat11-release (no repo).

**Review by scpeters — APPROVED** [review]

**Pull request merged**